rilldata/rill
# Fixing issues around dashboard state after an edit to the dashboard

## Checklist

## Summary
#### Issue addressed: 
Just after editing a dashboard the experience of using it was inconsistent.
1. The default state didnt match the new config.
2. The selections would not reflect changes unless the url state was cleared. This lead to user thinking the changes didnt apply.
3. Changing time controls and refreshing the page doesnt persist the url state.

#### Details:
1. Deriving the default proto state directly from the meta and the time range summary query to keep it in sync with dashboard config.
2. Clearing the dashboard state after an edit to the config.
3. Removed the forced reset of time range stuff.

## Steps to Verify
Issue 1 and 2
1. Create a dashboard with timestamp.
2. Navigate to the dashboard and change the selected time range and other settings.
4. Set the default_time_range to something that is valid within the dashboard.
5. Navigate to the dashboard and the new time range should be reflected as well as other settings should be reverted.

Issue 3
1. Change time range and other settings in the dashboard.
2. Refresh the page.
3. Time range selections should persist.
